viewer-prototype/src/browser/trace-server-preference.ts

```diff
@@ -0,0 +1,30 @@
+import { PreferenceSchema, PreferenceProxy, PreferenceScope } from '@theia/core/lib/browser';
+
+export const TRACE_PATH = 'trace-viewer.path';
+export const TRACE_PORT = 'trace-viewer.port';
+
+export const ServerSchema: PreferenceSchema = {
+    type: 'object',
+    properties: {
+        [TRACE_PATH]: {
+            'type': 'string',
+            'default': '',
+            'description': 'The path to trace-server executable, e.g.: /usr/bin/',
+        },
+        [TRACE_PORT]: {
+            'type': 'number',
+            'default': '',
+            'description': 'Specify the port on which you want to execute the server.',
```

Comment:
"you want", is it really up to me? Changing the port will have the server run on that port? It seems that if I put port 8082, everything is still on port 8080.

Reply:
Fixed. You can specify the port on which you want to execute the server and the listener will update the TraceServerUrl's port accordingly.